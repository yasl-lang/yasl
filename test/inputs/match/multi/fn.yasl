

fn matcher(...) {
   match ... {
   () {
       echo '()'
   }
   int {
       echo 'int'
   }
   int, int {
       echo 'int, int'
   }
   str, int {
       echo 'str, int'
   }
   bool, ... {
       echo 'bool, ...'
   }
   * {
       echo 'one'
   }
   ... {
       echo 'any'
   }
   }
}

matcher()
matcher(true, 10, 9, 8)
matcher(1)
matcher(1, 2)
matcher(1, 2, 3)
matcher('one', 2)
matcher('one')
matcher(1, 2, 3, 4)
