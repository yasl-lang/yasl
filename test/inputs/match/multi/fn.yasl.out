()
bool, ...
int
int, int
any
str, int
one
any
